pkgs/by-name/lr/lrcget/package.nix

```diff
@@ -0,0 +1,100 @@
+{ dbus
+, openssl
+, gtk3
+, webkitgtk
+, pkg-config
+, wrapGAppsHook3
+, fetchFromGitHub
+, buildNpmPackage
+, rustPlatform
+, lib
+, stdenv
+, copyDesktopItems
+, makeDesktopItem
+, alsa-lib
+}:
+let
+  pname = "lrcget";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "tranxuanthang";
+    repo = "lrcget";
+    rev = "${version}";
+    hash = "sha256-9Wtb8Y+cj1ec8vNxvvs3wso62TavNveHRh2Gfx7Br4w=";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  inherit version src pname;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoHash = "sha256-nhED/fSexFfZ2MezBR3YW5wKsYklRg9HiA3lvClkYjI=";
+
+  frontend = buildNpmPackage {
+    inherit version src;
+    pname = "lrcget-ui";
+    # FIXME: This is a workaround, because we have an git dependency node_modules/lrc-kit contains install scripts
+    # but has no lockfile, which is something that will probably break.
+    forceGitDeps = true;
+    distPhase = "true";
+    dontInstall = true;
+    # To fix `npm ERR! Your cache folder contains root-owned files`
+    makeCacheWritable = true;
+
+    npmDepsHash = "sha256-qQ5UMO3UuD6IvUveTRF35qTlGq5PMbxp1Q4UroDqVtk=";
+
+    # packageJSON = ./package.json;
+    postBuild = ''
+      cp -r dist/ $out
+    '';
+  };
+
+  # copy the frontend static resources to final build directory
+  # Also modify tauri.conf.json so that it expects the resources at the new location
+  postPatch = ''
+    mkdir -p frontend
+    cp -r ${frontend}/* frontend
+
+    substituteInPlace tauri.conf.json --replace-fail '"distDir": "../dist"' '"distDir": "./frontend"'
+  '';
+
+  nativeBuildInputs = [ pkg-config wrapGAppsHook3 copyDesktopItems ];
+  buildInputs = [ dbus openssl gtk3 webkitgtk alsa-lib ];
+
+  # Disable checkPhase, since the project doesn't contain tests
+  doCheck = false;
+
+  postInstall = ''
+    install -DT icons/128x128@2x.png $out/share/icons/hicolor/128x128@2/apps/lrcget.png
```

Comment:
Do we have `@2` in the freedesktop specification? I'm not sure

Reply:
am not sure, but `squirreldisk` (the package that I've taken as reference) did that, ref: https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/by-name/sq/squirreldisk/package.nix#L77